src/main/java/com/sap/cloudfoundry/client/facade/domain/CloudRoute.java

```diff
@@ -25,14 +26,49 @@ public int getAppsUsingRoute() {
     @Nullable
     public abstract String getPath();
 
-    @Override
-    public String getName() {
-        return getHost() + "." + getDomain();
-    }
+    @Nullable
+    public abstract Integer getPort();
+
+    public abstract String getUrl();
 
     @Override
     public CloudRoute derive() {
         return this;
     }
 
+    @Override
+    public String toString() {
+        return getUrl();
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getDomain().getName(), getHost(), getPath(), getPort());
+    }
+
+    @Override
+    public boolean equals(Object object) {
+        if (object == this) {
+            return true;
+        }
+        if (!(object instanceof CloudRoute)) {
+            return false;
+        }
+        var otherRoute = (CloudRoute) object;
+        var thisDomain = getDomain().getName();
+        var otherDomain = otherRoute.getDomain()
+                                    .getName();
+        return thisDomain.equals(otherDomain)
+                && areEmptyOrEqual(getHost(), otherRoute.getHost())
+                && areEmptyOrEqual(getPath(), otherRoute.getPath())
+                && Objects.equals(getPort(), otherRoute.getPort());
+    }
+
+    private static boolean areEmptyOrEqual(String lhs, String rhs) {
```

Comment:
_why static_ - because it doesn't use any instance variables and it's a utility method
_lhs and rhs_ - from C++ convention 😄 "left-hand side" and "right-hand side" (even though, yes, this isn't an operator overloading where left- and right-hand sides are present but still). Equivalent would be "left, right", "first, second", etc. The name isn't that important, so any names not being "string1" and "string2" would suffice. If you prefer other names, I will rename them.